### PR TITLE
Feature/list

### PR DESCRIPTION
**Pull request by dthmrphn**

### CODE REVIEW

File: src/test/tests.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+
+int test_pool(void);
+int test_list(void);
```
Comment (uladzislaudd) [inline src/test/tests.h:4]: empty line at EOF

**Comment by uladzislaudd** [comment]
@dthmrphn Please change file permissions, they should not be executable.

File: src/test/tests.h (hunk 1 above)
Comment (dthmrphn) [reply src/test/tests.h]: yep, added

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -16,6 +16,10 @@ add_library(mem STATIC
...
 )
 
-add_executable(test test/test_list.c)
+add_executable(test 
+    test/test_list.c
+    test/test_pool.c 
+    test/main.c
+)
 
 target_link_libraries(test PRIVATE mem)
```
Comment (uladzislaudd) [inline src/CMakeLists.txt:25]: empty line at EOF

File: src/test/test_list.c [hunk 3]
```diff
@@ -1,31 +1,70 @@
 #include "mem.h"
+#include "tests.h"
 
-#include <stdio.h>
+#include <stddef.h>
 #include <stdlib.h>
+#include <stdio.h>
```
Comment (uladzislaudd) [inline src/test/test_list.c:6]: nice "#include" grouping, but please sort them inside each group

Comment (uladzislaudd) [reply src/test/test_list.c:6]: @dthmrphn do it for every file

File: src/test/test_list.c [hunk 4]
```diff
@@ -1,31 +1,70 @@
...
-    .size = 1000
+    .size = sizeof(pool)
 };
 
-int main() {
+void set_buf(void *buf, size_t size) {
+    uint8_t *ptr = buf;
+    size_t i = 0;
+    while (i < size) {
+        ptr[i++] = (uint8_t)rand();
```
Comment (uladzislaudd) [inline src/test/test_list.c:19]: `*(int *)(ptr + i) = rand(); i += 4;` would be four times faster

**Review by uladzislaudd — APPROVED** [review]
LGFM! now please take a look at [this](https://github.com/ThrowTheSwitch/Unity). it is time to add auto tests.

File: src/test/test_list.c (hunk 4 above)
Comment (dthmrphn) [reply src/test/test_list.c:19]: yep, agreed, but this mostly stub function does nothing helpful

File: src/test/test_list.c (hunk 3 above)
Comment (dthmrphn) [reply src/test/test_list.c:6]: take this as advice for later use

**Pull request merged**